docs/rationale.md

```diff
@@ -54,6 +54,19 @@ Here are a few examples of things that are out of scope for prettier:
 * Adding/removing `{}` and `return` where they are optional.
 * Turning `?:` into `if then else`.
 
+## Can prettier add an option to do X?
+
+A lot of the current options in prettier come from the `recast` options of which prettier is a fork, and some of them were added later due to popular demand. Ideally we would not want options in prettier at all to keep printing things consistently and have less maintenance overhead but the JavaScript community has already chosen a few different coding styles people are used to:
+
+* semicolons/no semicolons
+* single/double quotes
+* spaces between brackets/parens
+* more recently, parens in arrow function arguments
+
+Supporting all of these options are a maintenance burden because it's hard to keep track when they start conflicting with each other; and if you want to have a comprehensive test suite, it would grow exponentially with the number of options you support.
```

Comment:
The reason eslint & co can get away with so many options is because their model is: when I see something bad, warn about it. Prettier on the other hand needs to be able to print every possible code you can throw at it using this option and make it look good. This is a --much-- harder problem.

We didn't even know if we could reasonably print JavaScript with no option at all, so it made sense to add as few options at the beginning.

Now that we know we can do it, it's still very challenging to add options and we're trying to only do so when a sizable amount of people are not willing to adopt prettier because of it. This way we can keep the project with a reasonable amount of complexity and when users want to use it, they are not overwhelmed by the number of options.